### PR TITLE
New Basemaps: Map Category - part 1 of 2

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

This PR is for the following sample categories:
- Map category (part 1 of 2)

Notes specific to this PR: 
- Create and Save Map: because getting the name from the new Basemaps has to wait until they load, there is a Basemap being created everytime it's changed at the moment. This has been noted as a possible future optimization to explore further when time allows, but for now, the sample works with the new styles / API key.

### General comments on Basemap updates:

It updates `Map` and/or `Basemap` Constructors as per the new design. This may include the following changes depending on the sample:
- Replace `Basemap.Type` with `BasemapStyle`
- If a lat/long/zoom was being set in the `Map` constructor, this is replaced with a `Viewpoint` being set on the `MapView`.
- Any Readme Updates identified

In addition, the samples that include the above changes, or other services requiring an API Key going forward, have the API Key  set within the sample. This will then work in conjunction with the Gradle Task being added in PR 593.

Once 593 is approved and merged into `dev` we will merge those changes into these PRs and then update all the Samples to `100.10.0-xxxx` and do a round of testing of the Samples with all the API Key / Basemap changes at that time. (We have left them as 100.9.0 (in most cases) for now to minimize merge conflicts in the `build.gradle` file).

In the meantime, the updates listed above can be reviewed. Thanks!

### CODE REVIEW

File: map/access-load-status/src/main/java/com/esri/samples/access_load_status/AccessLoadStatusSample.java
Comment (JonLavi) [inline map/access-load-status/src/main/java/com/esri/samples/access_load_status/AccessLoadStatusSample.java]: Maybe it'd be better to make this into 'plain english' by saying 'standard imagery basemap style'.
I might be wrong but possibly the enums are named in a format to have the 'imagery' appear first, followed by a qualifier (imagery, imagery standard, imagery labels, etc.). It's not 100% clear from the JavaDoc though, so I'll leave that to the phrasing specialists ;)

File: map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java
Comment (JonLavi) [inline map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java]: Could do plain english here as well?
```suggestion
      // setup a list view of basemaps styles
```

File: map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java
Comment (JonLavi) [inline map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java]: Sorry if this is a bit pedantic, but it's probably best naming the variable `basemapStyleListView` as well, as `List` is a different class with different implications (it's a data structure, rather than a UI element)

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (JonLavi) [inline map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: Maybe List View here as well? I see the other variables could also be named a bit better, so up to you if you want to include it in this PR or not
```suggestion
  private ListView<BasemapStyle> basemapStyleListView;
```

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (JonLavi) [inline map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: Small formatting suggestion, but up to you. I think these things look near when each entry is on it's own line, as I find it a lot easier to read:
```suggestion
    basemapStyleList.getItems().addAll(
	    BasemapStyle.ARCGIS_STREETS, 
	    BasemapStyle.ARCGIS_IMAGERY_STANDARD,
	    BasemapStyle.ARCGIS_TOPOGRAPHIC, 
	    BasemapStyle.ARCGIS_OCEANS);
```

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (JonLavi) [inline map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: It looks to me as if this variable is empty at this point? You have the same action on line 98, so it's probably ok to get rid of this line.

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (JonLavi) [inline map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: Perhaps it'd be easier to do this after line 85, to do all the setup steps for the list view at once (i.e. put in the values, set the cell factory)

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (JonLavi) [inline map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: I can't remember if there's an effor to not save single-use string variables (especially URLs), and just put them in-line. Meaning you could get rid of this line and just pop the url one line lower?
`ArcGISMapImageLayer worldElevation = new ArcGISMapImageLayer("https://sampleserver6.arcgisonline.com/arcgis/rest/services/WorldTimeZones/MapServer");`
Either way is good though!

**Review by JonLavi — APPROVED** [review]
Nice work @jenmerritt !
Just a bit of cleanup on `CreateAndSaveMapController.java` necessary. 
I've also put a few suggestions on variable naming, but you can pick and choose which ones you agree with 😄 
Approving so that you can move on to the second review once you've updated the file mentioned above.

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (jenmerritt) [reply map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: Ah - think I maybe reordered the MapView setup part or something so must have missed removing this. Thanks!

File: map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java
Comment (jenmerritt) [reply map/create-and-save-map/src/main/java/com/esri/samples/create_and_save_map/CreateAndSaveMapController.java]: Yeah we have been doing that I think you're right, I'll do that refactor

File: map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java
Comment (Rachael-E) [inline map/change-basemap/src/main/java/com/esri/samples/change_basemap/ChangeBasemapSample.java]: LOD 6 could be replaced by scale 9244648 (rounded down to 9240000 should be ok).

**Review by Rachael-E — APPROVED** [review]
looks good @jenmerritt , just one example of the scale to update.

**Comment by jenmerritt** [comment]
`dev` merged in (including API Key gradle tasks changes). Samples that have an API Key set have been tested running from the IDE.

Await merging these changes until release.

**Pull request merged**
